monai/apps/auto3d/data_utils.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+from typing import Any, Dict, List, Tuple, Union
+
+__all__ = ["recursive_getvalue", "recursive_setvalue", "recursive_getkey", "datafold_read"]
+
+
+def datafold_read(datalist: Union[str, Dict], basedir: str, fold: int = 0, key: str = "training") -> Tuple[List, List]:
+    """
+    Read a list of data dictionary `datalist`
+
+    Args:
+        datalist: the name of a JSON file listing the data, or a dictionary
+        basedir: directory of image files
+        fold: which fold to use (0..1 if in training set)
+        key: usually 'training' , but can try 'validation' or 'testing' to get the list data without labels (used in challenges)
+
+    Returns:
+        A tuple of two arrays (training, validation)
+    """
+
+    if isinstance(datalist, str):
+        with open(datalist) as f:
+            json_data = json.load(f)
+    else:
+        json_data = datalist
+
+    json_data = json_data[key]
+
+    for d in json_data:
+        for k, _ in d.items():
+            if isinstance(d[k], list):
+                d[k] = [os.path.join(basedir, iv) for iv in d[k]]
+            elif isinstance(d[k], str):
+                d[k] = os.path.join(basedir, d[k]) if len(d[k]) > 0 else d[k]
+
+    tr = []
+    val = []
+    for d in json_data:
+        if "fold" in d and d["fold"] == fold:
+            val.append(d)
+        else:
+            tr.append(d)
+
+    return tr, val
+
+
+def recursive_getvalue(dicts: Dict, keys: Union[str, List]) -> Any:
```

Comment:
For these recursive get/set value operations, maybe we can change to follow `ConfigParser` in the future:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/config_parser.py#L118
@wyli What do you think?

Thanks.

Reply:
this one handles the default=None case https://github.com/Project-MONAI/MONAI/blob/a9095d09d919e94c2c07aa74fc46c1e4f9530ecd/monai/bundle/config_parser.py#L165 agree that we can do it in the future